msteams-platform/bots/how-to/conversations/notification-bot-in-teams.md

```diff
@@ -0,0 +1,574 @@
+
+<details>
+
+<summary><b>Why Bad Request or Bad Argument error occurs when sending notification?</b></summary>
+
+If the notification installation doesn't match the bot ID or password, you can get a **Failed to decrypt conversation ID** error. One possible cause for this error is that the bot ID or password is changed due to cleaning local state or reprovisioning.
+
+You can resolve this by cleaning your notification storage. After cleaning, notify in Teams to reinstall your bot, and ensure that the new installation is up-to-date. Each stored notification installation is bound with one bot. If you're able to check your notification storage, its bot field should match the bot you're running such as the bot ID with the same GUID.
+
+> [!NOTE]
+> In case of local storage the default location is `.notification.localstore.json`.
+
+<br>
+
+</details>
+
+<details>
+
+<summary><b>Why notification target is lost after restarting or redeploying the bot app?</b></summary>
+
+Notification target connections are stored in the persistence storage. If you're using the default local file storage, Azure web app and Azure Functions clean up the local file during a restart or redeploy. You can also uninstall the bot from Teams, then install it to again add connections to the storage. It's recommended to use your own shared storage for production environment.
+
+<br>
+
+</details>
+
+<details>
+
+<summary><b>Why is undefined error returned when using the API findChannel()?</b></summary>
+
+You can encounter an undefined error, when the bot app is installed into other channels instead of the **General** channel. To fix this error, you can uninstall the bot app from Teams and redebug and relaunch it. After you've redebug and relaunched, ensure that the bot app is installed into the **General** channel.
+
+<br>
+
+</details>
+
+<details>
+
+<summary><b>Can I know all the targets where my bot is installed in and out of the notification project?</b></summary>
+
+There are Microsoft Graph APIs to list apps installed in a team, group, or chat. If required you need to iterate your team, group, or chat into an installed app to be targeted. In the notification project, it uses persistence storage to store installation targets. For more information, see [notification based on events](#notification-based-on-events).
+
+<br>
+
+</details>
+
+<details>
+
+<summary><b>How to customize the Azurite listening ports?</b></summary>
+
+If Azurite exits due to port in use, you can specify another listening port and update the connection string of `AzureWebJobsStorage` in `bot/local.settings.json`.
+
+<br>
+
+</details>
+
+<details>
+
+<summary><b>How to extend my notification bot to support command and response?</b></summary>
+
+If you need to extend your notification bot to support command and response, you can add Adaptive Cards to respond to external events. Follow the steps to add command and response [placeholder for add command and response in command bot in Teams].
+
```

Comment:
Can we add a link for `Follow the steps` links to command bot doc where we have those four steps.

Reply:
Yes this can be done.